### PR TITLE
refactor(netcheck): minor code improvements

### PR DESCRIPTION
**Pull request by divagant-martian**
- removes an unnecessary clone
- makes sure the port mapping probe result is used entirely by destructuring it
- changes a log to be emitted only if it's relevant

### CODE REVIEW

**Review by dignifiedquire — APPROVED** [review]

File: iroh-net/src/hp/netcheck.rs [hunk 1]
```diff
@@ -693,14 +693,16 @@ impl ReportState {
...
+            debug!("aborting {} probes, already done", probes.len());
+        }
         drop(probes);
 
         if let Some(hair_pin) = self.wait_hair_check().await {
             self.report.write().await.hair_pinning = Some(hair_pin);
         }
 
-        if !skip_external_network && port_mapper.is_some() {
+        if do_port_map {
```
Comment (flub) [inline iroh-net/src/hp/netcheck.rs:705]: so we start portmapping anyway as long as we're allows to use an external network, but don't always bother waiting for it?  what's the logic for that?

**Review by flub — APPROVED** [review]

File: iroh-net/src/hp/netcheck.rs (hunk 1 above)
Comment (divagant-martian) [reply iroh-net/src/hp/netcheck.rs:705]: I'm not sure what you are referring to. The logic is akin to what was in here already. In reality we could just do `port_mapping.inner.is_some()`

Comment (flub) [reply iroh-net/src/hp/netcheck.rs:705]: yeah, i know the logic didn't change.  i was only trying to understand myself why it was this way and given you modified it maybe you knew...

Comment (divagant-martian) [reply iroh-net/src/hp/netcheck.rs:705]: ah I see. My understanding is that we always finish it actually. At least that's my understanding from tailscale's code. Code as is pretends to do it with the line just below this one without actually polling the future. But in theory, yes, we should finish the probe

**Comment by dignifiedquire** [comment]
ci seems sad

**Comment by divagant-martian** [comment]
> ci seems sad

dump report failed + cargo deny, neither seems related to the changes. Made the sims one run again to be sure

**Comment by dignifiedquire** [comment]
> cargo deny,

trying to fix this here: https://github.com/n0-computer/iroh/pull/1141

**Review by flub — APPROVED** [review]

**Pull request closed without merging**